vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/LinkCrawler.java

```diff
@@ -69,14 +75,50 @@ public void visit(Page page)
         }
     }
 
-    private static boolean isAllowedUrl(Page referringPage, WebURL url)
+    private boolean isAllowedUrl(Page referringPage, WebURL url)
+    {
+        return isNotFileWithForbiddenExtension(url)
+                && isFromTheSameSite(referringPage.getWebURL().getURL(), url.getURL());
+    }
+
+    private boolean isNotFileWithForbiddenExtension(WebURL url)
+    {
+        String fileExtensionFromUrl = getExtension(url.getPath()).toLowerCase();
+        boolean notExcludedExtension = isNotExcludedExtension(fileExtensionFromUrl);
+        if (!notExcludedExtension)
+        {
+            LINK_CRAWLER_LOGGER.info("Skip crawling for URL {}", url.getURL());
+        }
+        return notExcludedExtension;
+    }
+
+    @Override
+    protected void onRedirectedStatusCode(Page page)
+    {
+        URI redirectedUri = UriUtils.createUri(page.getRedirectedToUrl());
+        UriUtils.removeQuery(redirectedUri);
+        String fileExtensionFromRedirectedUrl = getExtension(redirectedUri.getPath()).toLowerCase();
+
+        if (!isNotExcludedExtension(fileExtensionFromRedirectedUrl))
+        {
+            String url = page.getWebURL().getURL();
+            urlsWithInvalidRedirects.add(url);
```

Comment:
just to confirm: the parsing of the redirected page (even if it's forbidden) will still be performed, right?

Reply:
> just to confirm: the parsing of the redirected page (even if it's forbidden) will still be performed, right?

The redirected page will pass the check "shouldVisit" just like any other links, if the check fails - the page will not be parsed

Reply:
ok, in this case how do invalid redirected pages appear in `linkCrawlerData` (in other words why do we need this line: `linkCrawlerData.getAbsoluteUrls().removeAll(urlsWithInvalidRedirects);`?

Reply:
> ok, in this case how do invalid redirected pages appear in `linkCrawlerData` (in other words why do we need this line: `linkCrawlerData.getAbsoluteUrls().removeAll(urlsWithInvalidRedirects);`?

There must be some misunderstanding here : )
By "redirected page" I mean the page we redirected to
This block of code with "removeAll" removes the page from which we redirected